store/driver/txn/txn_driver.go

```diff
@@ -114,8 +133,16 @@ func (txn *tikvTxn) Delete(k kv.Key) error {
 }
 
 func (txn *tikvTxn) Get(ctx context.Context, k kv.Key) ([]byte, error) {
-	data, err := txn.KVTxn.Get(ctx, k)
-	return data, derr.ToTiDBErr(err)
+	val, err := txn.GetMemBuffer().Get(ctx, k)
+	if kv.ErrNotExist.Equal(err) {
+		val, err = txn.GetSnapshot().Get(ctx, k)
+	}
+
+	if len(val) == 0 {
+		return nil, kv.ErrNotExist
```

Comment:
If `err` is some other type error rather than `ErrNotExist`, it's mistakenly returned as `ErrNotExist`

Reply:
My fault... It seems no cases are covering it, we should have more test cases... I'll add some later